doc/user_guide/index.rst

```diff
@@ -9,7 +9,7 @@ User Guide
    installation
    run
    output
-   message-control
+   messages/index
```

Comment:
Don't we want `messages` here so that we get all sub-headers as links as well?

Reply:
Maybe something I don't know about sphinx but when using ``messages/`` here I get : ``/home/pierre/pylint/doc/user_guide/index.rst:5: WARNING: toctree contains reference to nonexisting document 'user_guide/messages'
`` during build ?

Reply:
Ah, the solution is different: add a hidden `toctree` to `messages/index.rst`. See difference between certain headers on:
https://pylint--6628.org.readthedocs.build/en/6628/user_guide/index.html

Reply:
Unless I'm mistaken there's already a hidden toctree in message/index ?

Reply:
https://github.com/PyCQA/pylint/pull/6628/files#diff-02baef687b1c6a68cf7f524437fced9e405fe591b8dff8e7736b038aecced7dcR9

Reply:
I checked, for the headers in which the sub-headers do show up the `toctree` is actually visible and not hidden.

I'm not sure we want to make the `toctree` for messages visible so I guess this will have to make do? Or is there a `overwrite-subhearder-visibility` option for `toctree`?

Reply:
I'm not sure about what we want to do, maybe we can delay this to the last doc reorganization MR ?